contrib/packs/actions/pack_mgmt/unload.py

```diff
@@ -109,9 +109,22 @@ def _unregister_pack(self, pack):
         return True
 
     def _delete_pack_db_object(self, pack):
+        pack_db = None
+
+        # 1. Try by ref
         try:
-            pack_db = Pack.get_by_name(value=pack)
+            pack_db = Pack.get_by_ref(value=pack)
         except StackStormDBObjectNotFoundError:
+            pack_db = None
+
+        # 2. Try by name (here for backward compatibility)
+        if not pack_db:
```

Comment:
For just in case aka backward compatibility I left this code in place.

Plan to remove it in 2.1 or similar.